### PR TITLE
fix: The 'indentSize' in the Table component couldn't be zero value

### PR DESCRIPTION
**Pull request by OmriGM**
- Changed to indentSize prop logic of the Table component to also include zero values

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/25798

### 💡 Background and solution
Before:

 ```mergedExpandable.indentSize = mergedExpandable.indentSize || indentSize || 15; ```

After:

 ```
 if (mergedExpandable.indentSize === null || mergedExpandable.indentSize === undefined) {
    mergedExpandable.indentSize = indentSize !== null && indentSize !== undefined ? indentSize : 15;
  }
```

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Table `indentSize` couldn't be zero value. |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by 07akioni** [comment]
CI failed

**Comment by OmriGM** [comment]
@07akioni  When I run `npm test -- -w 1` (like in the failing CI tests) locally all the tests passes, any idea?

File: components/cascader/__tests__/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -1518,11 +1512,8 @@ exports[`Cascader should render not found content 1`] = `
       motionEnter={false}
       motionLeave={false}
       motionName="slide-up"
-      onAppearStart={[Function]}
```
Comment (afc163) [inline components/cascader/__tests__/__snapshots__/index.test.js.snap:1521]: You should't update this, please run `npm test -- -u` after reinstall `node_modules`.

**Comment by afc163** [comment]
Could you add a test case?

File: components/table/Table.tsx [hunk 2]
```diff
@@ -409,7 +404,9 @@ function Table<RecordType extends object = any>(props: TableProps<RecordType>) {
   }
 
   // Indent size
-  mergedExpandable.indentSize = mergedExpandable.indentSize || indentSize || 15;
+  if (mergedExpandable.indentSize === null || mergedExpandable.indentSize === undefined) {
```
Comment (07akioni) [inline components/table/Table.tsx:407]: maybe use `typeof xxx === 'number'` is a more evident solution?

Comment (OmriGM) [reply components/table/Table.tsx]: Updated thanks to your comment

**Pull request merged**